PatternPal/PatternPal.SyntaxTree/Abstractions/Root/IRoot.cs

```diff
@@ -6,49 +6,63 @@
 
 namespace PatternPal.SyntaxTree.Abstractions.Root
 {
+    /// <summary>
+    /// A root represents a compiled file in the syntax graph.
+    /// </summary>
     public interface IRoot : IEntitiesContainer, INamespaceContainer, IUsingContainer
     {
         /// <summary>
-        ///     Get the filepath of the source file where this node is found in
+        /// Gets the filepath of the source file where this node is found in.
         /// </summary>
-        /// <returns>Filepath of the source</returns>
         string GetSource();
 
         IEnumerable<Relation> GetRelations(INode node, RelationTargetKind type);
     }
 
+    /// <summary>
+    /// Represents a node which has namespaces.
```

Comment:
This comment is a little unclear to me, do you mean to say that this interface represents things which can be contained in a namespace, or things which can contain a namespace? Without additional knowledge, this is difficult to deduce.